cmd/webhook.go

```diff
@@ -229,41 +237,46 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 
 	filerBucketList := make([]string, 0)
 
-	for _, secret := range secretList.Items {
-		// Check for secrets having filer-conn-secret
-		if strings.Contains(secret.Name, "filer-conn-secret") {
-			// Obtain the name of the filer to further unique mounts and organization
-			filerNameList := strings.Split(secret.Name, "-")
-			filerName := "error" // should not happen
-			if len(filerNameList) > 1 {
-				filerName = filerNameList[0]
-			}
-
+	// shareList.Data is a map[string]string
+	// https://goplay.tools/snippet/zUiIt23ZYVK
+	var shareList []string
+	for svmName := range svmShareList.Data {
+		// Retrieve the associated secret with the svm
+		secret, err := clientset.CoreV1().Secrets(pod.Namespace).Get(context.Background(),
```

Comment:
Im starting to think just looking at this, if we should add a label to the user secrets if possible to store the legit "svmName" coming from the masterfiler list. And then we can filter on that instead of the name. 
Although honestly, as long as the svmname/vserver value stays unique in the master list, then going by the name will be fine. So what im saying is, this is just me saying things.